From 863b3a5f521e5983619925bd7078386c2413b31b Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Tue, 13 Aug 2024 19:53:31 +0200
Subject: [PATCH 1/3] wifi: mt76: mt7915: set MT76_MCU_RESET early in
 mt7915_mac_full_reset

This avoids running into unnecessary timeouts waiting for MCU responses

Signed-off-by: Felix Fietkau <nbd@nbd.name>
---
 mt7915/mac.c | 1 +
 1 file changed, 1 insertion(+)

--- a/mt7915/mac.c
+++ b/mt7915/mac.c
@@ -1448,6 +1448,7 @@ mt7915_mac_full_reset(struct mt7915_dev
 
 	dev->recovery.hw_full_reset = true;
 
+	set_bit(MT76_MCU_RESET, &dev->mphy.state);
 	wake_up(&dev->mt76.mcu.wait);
 	ieee80211_stop_queues(mt76_hw(dev));
 	if (ext_phy)
